task_scheduler.py

```diff
@@ -0,0 +1,87 @@
+from datetime import datetime as dt
+
+class Task:
+    def __init__(self, name: str, duration: int, deadline: str, completed = False) -> None:
+        # tasks begin as soon as they are added
+        self.name = name
+        self.duration = duration
+        # convert deadline from 'YYYY-MM-DD HH:MM' to datetime object
+        self.deadline = dt.strptime(deadline, '%Y-%m-%d %H:%M')
+        self.completed = completed
+
+class TaskScheduler:
+
+    STATUS_MESSAGE = {
+        
+    }
+
+    def __init__(self, tasks = None) -> None:
+        if tasks is None:
+            tasks = {}
+        self.tasks = tasks
+
+    def validate_task(self, task = None):
+        if task is None:
+            raise ValueError('No task by that name')
+
+    def add_task(self, name: str, duration: int, deadline: str):
+        added_task = Task(name, duration, deadline)
+        self.tasks[name] = added_task
+
+    def check_status(self, name: str):
+        current_task = self.tasks.get(name, None)
+        try:
+            self.validate_task(current_task)
+            if current_task.completed is False:
+                return 'In progress'
+            elif current_task.completed is True:
+                return 'Completed'
+            else:
+                return 'Not started'
```

Comment:
You're tangling your concerns too much. Why should the `TaskScheduler` be responsible for determining the text form of the task's status? This logic should be moved to the `Task` class. Have a `current_task.get_status()` method that does this.